packages/core/src/internal/validation/futures/validateArtifactContractAt.ts

```diff
@@ -16,9 +17,11 @@ export async function validateArtifactContractAt(
   /* stage two */
 
   if (isModuleParameterRuntimeValue(future.address)) {
-    const param =
-      deploymentParameters[future.address.moduleId]?.[future.address.name] ??
-      future.address.defaultValue;
+    const param = resolvePotentialModuleParameterValueFrom(
```

Comment:
I refactored out the logic of module parameter resolution given deployed parameters and a ModuleParameterRuntimeValue.